src/self-managed-debug/src/system_catalog_dumper.rs

```diff
@@ -674,110 +848,195 @@ impl<'n> SystemCatalogDumper<'n> {
 
         Ok(Self {
             context,
-            pg_client: Arc::new(pg_client),
-            _cluster_replicas: cluster_replicas,
+            pg_client: Arc::new(Mutex::new(pg_client)),
+            pg_tls,
+            cluster_replicas,
             _pg_conn_handle: handle,
         })
     }
 
-    pub fn dump_relation(&self, relation: &Relation) -> JoinHandle<()> {
-        let start_time = self.context.start_time.clone();
-        let pg_client = Arc::clone(&self.pg_client);
+    pub async fn dump_relation(
+        &self,
+        relation: &Relation,
+        cluster_replica: Option<&ClusterReplica>,
+    ) -> Result<(), anyhow::Error> {
+        info!(
+            "Copying relation {}{}",
+            relation.name,
+            cluster_replica.map_or_else(|| "".to_string(), |replica| format!(" in {}", replica))
+        );
+
+        let start_time = self.context.start_time;
+        let pg_client = &self.pg_client;
+
         let relation_name = relation.name.to_string();
-        let relation_category = relation.category.clone();
-
-        task::spawn(|| "dump-relation", async move {
-            if let Err(err) = retry::Retry::default()
-                .max_duration(Duration::from_secs(60))
-                .retry_async_canceling(|retry_state| {
-                    if retry_state.i > 0 && retry_state.next_backoff.is_some() {
-                        info!(
-                            "Retrying {} in {:?}",
-                            relation_name,
-                            retry_state.next_backoff.unwrap()
-                        );
-                    }
 
-                    let start_time = start_time.clone();
-                    let pg_client = Arc::clone(&pg_client);
-                    let relation_name = relation_name.clone();
-                    let relation_category = relation_category.clone();
+        let mut pg_client_lock = pg_client.lock().await;
```

Comment:
Can you document somewhere in this crate why we don't want to run these queries in parallel, and we'd prefer serial execution?